sphinx_extensions/doxysphinx_theme/__init__.py

```diff
@@ -0,0 +1,43 @@
+# =====================================================================================
+#  C O P Y R I G H T
+# -------------------------------------------------------------------------------------
+#  Copyright (c) 2023 by Robert Bosch GmbH. All rights reserved.
+#
+#  Author(s):
+#  - Aniket Salve, Robert Bosch GmbH
+# =====================================================================================
+
+"""Doxysphinx themes extension.
+
+This extensions add custom javascript files depending on theme.
+Currently only supporting Book and RTD theme.
+"""
+
+import pathlib
+
+from sphinx.application import Sphinx
+from sphinx.config import Config
+
+
+def setup(app: Sphinx):
+    """Setups up the replacer extension."""
+    app.connect("config-inited", doxysphinx_theme_extension)
+    return {"parallel_read_safe": True, "parallel_write_safe": True, "version": "0.1.0"}
+
+
+def doxysphinx_theme_extension(app: Sphinx, config: Config):
+    """Add custom javascript files specific to theme and set theme options.
+
+    :param app: Sphinx app
+    :param config: Sphinx config
+    """
+    current_file_path = pathlib.Path(__file__).parent.resolve()
+
+    config.html_static_path.append(str(current_file_path) + "/_static/")
+
+    if config.html_theme == "sphinx_book_theme":
+        app.add_js_file("js/customize-navbar-book.js")
+
+    elif config.html_theme == "sphinx_rtd_theme":
+        app.add_js_file("js/customize-navbar-rtd.js")
+        config.html_theme_options["collapse_navigation"] = False
```

Comment:
maybe a prior check with a warning message would be nice saying "We need to overwrite your collapse_navigation setting and set it to false" in case a user has set that by intention and is then wondering why it's not applied.. mhmm however if collapse_navigation = True is the default we'll always get the warning (not good) - what do you think?

Reply:
We can do a check if flag was true before then show warning.  (Not sure if we are over engineering it then)

Also when we say warning, you mean console warning right? Not pop-up in VS Code